src/basic/fileio.c

```diff
@@ -780,6 +781,7 @@ static int merge_env_file_push(
 
         if (!env_name_is_valid(key)) {
                 log_error("%s:%u: invalid variable name \"%s\", ignoring.", strna(filename), line, key);
+                free(value);
```

Comment:
BOMs are a gruesome concept I must say, I am not sure we should really care too much about them...